packages/calcite-components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -396,6 +397,10 @@ export class TabNav {
     if (totalVisibleTabTitles === 1 && tabTitles[visibleTabTitlesIndices[0]].closable) {
       tabTitles[visibleTabTitlesIndices[0]].closable = false;
       this.selectedTabId = visibleTabTitlesIndices[0];
+
+      if (selectionModified) {
+        tabTitles[visibleTabTitlesIndices[0]].click(); // trigger selection-related events since they were modified after user interaction
```

Comment:
nitpick: It seems a little odd to be handling this using a click since a click will bubble up and could potentially trigger other unknown things higher in the DOM.

Maybe in the future we could refactor this to handle the use case without having to trigger a click of an element? Just throwing thoughts out there :) I know we do use `.click()` in other places but its something we should think about possibly.

Reply:
100% agreed. I'll update this PR to add an internal method to only trigger the tab-related events and will create a refactor issue to introduce a tabs controller that `tabs` and its structure could share to make this cleaner. There's a lot of eventing going back and forth and it's getting quite complex.

Reply:
Internal method added and `click` no longer used. Can you take another look? 🙇‍♂️